Sources/ForageSDK/ForageSDK.swift

```diff
@@ -38,10 +39,21 @@ public class ForageSDK {
             return
         }
         self.environment = config.environment
-        LDManager.shared.initialize(self.environment)
-        // TODO: Maybe move this shared logger call!
+        // ForageSDK.shared.environment is not set
+        // until the end of this initialization
+        // so we have to provide the environment from the config
+        let logger = DatadogLogger(
+            ForageLoggerConfig(
+                environment: config.environment,
+                prefix: ""
+            )
+        )
+        self.logger = logger
+        LDManager.shared.initialize(self.environment, logger: logger)
         VGSCollectLogger.shared.disableAllLoggers()
-        self.service = LiveForageService()
+        
+        let provider = Provider(logger: logger)
+        self.service = LiveForageService(provider: provider, logger: logger)
```

Comment:
love this! Instantiate a logger at the top level and pass as a dependency down to keep services pure. Great work!